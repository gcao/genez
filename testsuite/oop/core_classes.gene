#!/usr/bin/env gene run

# Test: Core class hierarchy
(print "Int class: " Int)
(print "String class: " String)
(print "Array class: " Array)
(print "Map class: " Map)

# Everything inherits from Any
(print "Any class: " Any)