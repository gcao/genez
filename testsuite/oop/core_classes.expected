Int class:  Class: Int
String class:  Class: String
Array class:  Class: Array
Map class:  Class: Map
Any class:  Class: Any